packages/fxa-settings/src/components/FormPasswordWithBalloons/index.tsx

```diff
@@ -152,23 +155,53 @@ export const FormPasswordWithBalloons = ({
     if (getValues('newPassword') !== '' && !errors.newPassword) {
       hideNewPwdBalloon();
     }
+    if (getValues('confirmPassword') !== getValues('newPassword')) {
+      setPasswordMatchErrorText(localizedPasswordMatchError);
+    } else if (!formState.isValid) {
+      // Try to retrigger
+      trigger('confirmPassword');
+    }
   };
 
   const onBlurConfirmPassword = useCallback(() => {
     passwordFormType === 'signup' &&
       isConfirmPwdBalloonVisible &&
       hideConfirmPwdBalloon();
 
-    getValues('confirmPassword') !== getValues('newPassword') &&
+    if (getValues('confirmPassword') !== getValues('newPassword')) {
       setPasswordMatchErrorText(localizedPasswordMatchError);
+    } else if (!formState.isValid) {
+      trigger('newPassword');
+    }
   }, [
     getValues,
     hideConfirmPwdBalloon,
     isConfirmPwdBalloonVisible,
     localizedPasswordMatchError,
     passwordFormType,
     setPasswordMatchErrorText,
+    formState,
+    trigger,
   ]);
+  const onChangePassword = (inputName: string) => {
+    if (inputName === 'confirmPassword') {
+      setHasEditedConfirmPwd(true);
+    }
+
+    if (!hasEditedConfirmPwd) {
+      return;
+    }
+
+    if (getValues('confirmPassword') !== getValues('newPassword')) {
+      setPasswordMatchErrorText(localizedPasswordMatchError);
+    } else {
+      setPasswordMatchErrorText('');
+    }
+
+    if (!formState.isValid) {
+      trigger(inputName);
+    }
```

Comment:
I've tested locally, and the only change I would recommend is the following:

```suggestion
    trigger('confirmPassword');
```

- We should only care about triggering the password match check from 'confirmPassword'
- We can remove the conditional, we should check for a change in validity whether or not the form state was previously valid/invalid.